lib/pwauth/okta/impl.go

```diff
@@ -0,0 +1,68 @@
+package okta
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/Symantec/Dominator/lib/log"
+	"net/http"
+)
+
+const (
+	authPath           = "/api/v1/authn"
+	authEndpointFormat = "https://%s.okta.com" + authPath
+)
+
+type loginDataType struct {
+	Password string `json:"password,omitempty"`
+	Username string `json:"username,omitempty"`
+}
+
+type responseType struct {
+	Status string `json:"status,omitempty"`
+}
+
+func newPublicAuthenticator(oktaDomain string, logger log.Logger) (
+	*PasswordAuthenticator, error) {
+	return &PasswordAuthenticator{
+		authnURL: fmt.Sprintf(authEndpointFormat, oktaDomain),
+		logger:   logger,
+	}, nil
+}
+
+func (pa *PasswordAuthenticator) passwordAuthenticate(username string,
+	password []byte) (bool, error) {
+	loginData := loginDataType{Password: string(password), Username: username}
+	body := &bytes.Buffer{}
+	encoder := json.NewEncoder(body)
+	encoder.SetIndent("", "    ") // Make life easier for debugging.
+	if err := encoder.Encode(loginData); err != nil {
+		return false, err
+	}
+	if req, err := http.NewRequest("POST", pa.authnURL, body); err != nil {
+		return false, err
+	} else {
```

Comment:
There is no need to put this within else statements, just complicated the readout. It should be
if req, err: ==... {
     return false, err
}
//add headers
if resp, err := http.DefaultClient.Do( .... {
    return false, err
}
if resp.StatusCode ==  http.StatusUnauthorized {
    return false, nil
}
if resp.Code != StatusOK {
    return false, fmt.Errorf(....)
}
decoder := ....

Reply:
Your sample code won't work because the `req` and `resp` variables are now being used outside their scope. To make it work, you would need:
`req, err := blah()
if err != nil {
}
blag(req)
`
As you can see, it adds more lines. The way I've done it is more idiomatic Go, from what I've seen.

Reply:
I strongly prefer those extra lines, the flow control is much more cleaner (which is very important on security sensitive code code).

Reply:
Done. PTAL.